ee/packages/pdf-worker/src/worker.spec.ts

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs';
+
+import { PdfWorker } from './index';
+import {
+	bigConversationData,
+	dataWithASingleMessageButAReallyLongMessage,
+	dataWithMultipleMessagesAndABigMessage,
+	dataWithASingleMessageAndAnImage,
+} from './worker.fixtures';
+
+const streamToBuffer = async (stream: NodeJS.ReadableStream) => {
+	const chunks: (string | Buffer)[] = [];
+	for await (const chunk of stream) {
+		chunks.push(chunk);
+	}
+
+	return Buffer.concat(chunks as Buffer[]);
+};
+
+const pdfWorker = new PdfWorker('chat-transcript');
+
+describe('PdfWorker', () => {
+	it('should fail to instantiate if no mode is provided', () => {
+		// @ts-expect-error - testing
+		expect(() => new PdfWorker('')).toThrow();
+	});
+	it('should fail to instantiate if mode is invalid', () => {
+		// @ts-expect-error - testing
+		expect(() => new PdfWorker('invalid')).toThrow();
+	});
+	it('should properly instantiate', () => {
+		const newWorker = new PdfWorker('chat-transcript');
+
+		expect(newWorker).toBeInstanceOf(PdfWorker);
+		expect(newWorker.mode).toBe('chat-transcript');
+	});
+	it('should generate a pdf transcript for a big bunch of messages', async () => {
+		const stream = await pdfWorker.renderToStream({ data: bigConversationData });
+		const buffer = await streamToBuffer(stream);
+
+		expect(buffer).toBeTruthy();
```

Comment:
In which case this would be falsy? (Same for the others) 🤔

Reply:
actually this just checks that buffer "exists". If the generation fails, the process would just be stuck 😬 with no errors thrown so if we have a buffer here, is because we're good

Reply:
But the `streamToBuffer` never returns a falsy value, right?